### PR TITLE
Add design guidelines

### PR DESCRIPTION
**Pull request by colebemis**
This PR adds usage and design guidelines to the Octicons website.

### Preview link 👀 

* [Design guidelines](https://octicons-git-guidelines.primer.vercel.app/octicons/guidelines/design)

### References

- [Initial google doc](https://docs.google.com/document/d/15DsYasuwS8R6J3X70yyTsreTz-I1f0WVyFwyURb-dtM/edit#)
- [Nucleo icon guidelines](https://blog.nucleoapp.com/nucleo-icon-guidelines-introduction-70092f8b4697)
- [Carbon icon guidelines](https://www.carbondesignsystem.com/guidelines/icons/contribute/#contribution-process)
- [Material icon guidelines](https://material.io/design/iconography/system-icons.html#design-principles)
- [Feather icon guidelines](https://github.com/feathericons/feather/issues/171#issuecomment-455356985)

### CODE REVIEW

File: docs/content/guidelines/design.mdx [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+## Grid sizes
+
+Design two versions of each icon: a 16px version and a 24px version. 
+
+<Grid gridTemplateColumns={['1fr', null, null, null, '1fr 1fr']} gridGap={4}>
+    <div>
+        <Flex bg="gray.1" justifyContent="center">
+            <img width="400" alt="" src="https://compai.pub/v1/png/41d55c2894e91bd7d0594d005ce51e2471292973343c01dcbbc271c34aa2d702"/>
+        </Flex>
+        <Caption>16x16 grid</Caption>
```
Comment (yaili) [inline docs/content/guidelines/design.mdx:19]: Wondering if this caption is enough or if we should make sure there's alt tags in all the images.

File: docs/content/guidelines/design.mdx [hunk 2]
```diff
@@ -0,0 +1,300 @@
+---
+title: Design guidelines
+---
+
+import {Caption, Do, Dont} from '@primer/gatsby-theme-doctocat'
+import {Grid, Flex, Box} from '@primer/components'
+
+These are guidelines to use as a reference when designing new Octicons. If you're looking for guidance on how to use Octicons in your design, see our [Usage guidelines](/guidelines/usage).
```
Comment (yaili) [inline docs/content/guidelines/design.mdx:8]: Should _octicons_ be lowercase if in the middle of a sentence?

File: docs/content/guidelines/design.mdx [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+Here's how it works:
+1. Select the icon frames you want to commit. Make sure the frames are either 16x16 or 24x24 and that you've outlined all strokes.
+1. Open the Octicons Push plugin.
+1. Select the branch you want to commit to. You can choose an existing branch or create a new branch.
+1. Press "Commit." The plugin will then export, commit, and push the selected icons to the branch you chose. If you chose to create a new branch, the plugin will give you a link to where you can start a new pull request with your branch.
+
+
+![demo](https://user-images.githubusercontent.com/4608155/77948730-b1a24600-727a-11ea-9c39-040be9a12963.gif)
+
+### Review questions
```
Comment (yaili) [inline docs/content/guidelines/design.mdx:294]: Will there be a sentence here like: _Before starting to design a new octicon, consider the following:_

File: docs/content/guidelines/design.mdx [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+
+### Creating a pull request
+
+If you work at GitHub, you can use the [Octicons Push](https://www.figma.com/community/plugin/825432045044458754/Octicons-Push) Figma plugin to start an Octicons pull request from Figma.
+
+Here's how it works:
+1. Select the icon frames you want to commit. Make sure the frames are either 16x16 or 24x24 and that you've outlined all strokes.
+1. Open the Octicons Push plugin.
+1. Select the branch you want to commit to. You can choose an existing branch or create a new branch.
+1. Press "Commit." The plugin will then export, commit, and push the selected icons to the branch you chose. If you chose to create a new branch, the plugin will give you a link to where you can start a new pull request with your branch.
```
Comment (yaili) [inline docs/content/guidelines/design.mdx:289]: What happens next? Will it be reviewed?

**Review by yaili — COMMENTED** [review]
Added a few comments. Looking good! 👍🏼

File: docs/content/guidelines/design.mdx (hunk 1 above)
Comment (colebemis) [reply docs/content/guidelines/design.mdx:19]: @emplums explained to me that if we have descriptive captions we should leave the `alt` attribute empty. From our [Caption docs](https://primer.style/doctocat/components/caption):
> Be sure to provide enough detail in your caption so users with assistive technology have adequate context. An empty alt tag alt="" should be used on the img in cases where the caption describes the image, thus making the image decorative in purpose.

File: docs/content/guidelines/design.mdx (hunk 2 above)
Comment (colebemis) [inline docs/content/guidelines/design.mdx:8]:
```suggestion
These are guidelines to use as a reference when designing new octicons. If you're looking for guidance on how to use octicons in your design, see our [Usage guidelines](/guidelines/usage).
```

Comment (colebemis) [reply docs/content/guidelines/design.mdx]: Yeah, I think you're right.

File: docs/content/guidelines/design.mdx [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+
+
+![demo](https://user-images.githubusercontent.com/4608155/77948730-b1a24600-727a-11ea-9c39-040be9a12963.gif)
+
+### Review questions
+
+- Where will this icon be used in the context of GitHub UI?
+- Is an icon necessary in that context?
```
Comment (colebemis) [inline docs/content/guidelines/design.mdx:297]:
```suggestion
### Review questions

Here are a few questions we'll ask when reviewing new octicons. Keep these in mind as you're designing:

- Where will this icon be used in the context of GitHub UI?
- Is an icon necessary in that context?
```

File: docs/content/guidelines/design.mdx (hunk 3 above)
Comment (colebemis) [reply docs/content/guidelines/design.mdx:294]: Good idea. I added a short intro for this section. How does this sound?

> Here are a few questions we'll ask when reviewing new octicons. Keep these in mind as you're designing:

File: docs/content/guidelines/design.mdx (hunk 4 above)
Comment (colebemis) [inline docs/content/guidelines/design.mdx:289]:
```suggestion
1. Press "Commit." The plugin will then export, commit, and push the selected icons to the branch you chose. If you chose to create a new branch, the plugin will give you a link to where you can start a new pull request with your branch.

After you create a pull request, a member of the [design systems](https://primer.style/about/#team/) team will triage and review your contribution.
```

Comment (colebemis) [reply docs/content/guidelines/design.mdx:289]: Updated ✅

File: docs/src/@primer/gatsby-theme-doctocat/nav.yml [hunk 6]
```diff
@@ -13,3 +15,8 @@
       url: /packages/react
     - title: Styled System
       url: /packages/styled-system
+- title: Guidelines
+  url: /guidelines
+  children:
+    - title: Design
+      url: /guidelines/design
```
Comment (colebemis) [inline docs/src/@primer/gatsby-theme-doctocat/nav.yml:22]:
```suggestion
    - title: Usage
      url: /guidelines/usage
    - title: Design
      url: /guidelines/design
```

**Review by yaili — APPROVED** [review]
👍🏼  Looks good. Just needs the gif alt, as discussed. 🙂

File: docs/content/guidelines/design.mdx [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+Here's how it works:
+1. Select the icon frames you want to commit. Make sure the frames are either 16x16 or 24x24 and that you've outlined all strokes.
+1. Open the Octicons Push plugin.
+1. Select the branch you want to commit to. You can choose an existing branch or create a new branch.
+1. Press "Commit." The plugin will then export, commit, and push the selected icons to the branch you chose. If you chose to create a new branch, the plugin will give you a link to where you can start a new pull request with your branch.
+
+After you create a pull request, a member of the [design systems](https://primer.style/about/#team/) team will triage and review your contribution.
+
+
+![demo](https://user-images.githubusercontent.com/4608155/77948730-b1a24600-727a-11ea-9c39-040be9a12963.gif)
```
Comment (colebemis) [inline docs/content/guidelines/design.mdx:294]:
```suggestion
![demo showing how to create a pull request using the Octicons Push Figma plugin](https://user-images.githubusercontent.com/4608155/77948730-b1a24600-727a-11ea-9c39-040be9a12963.gif)
```

**Comment by ashygee** [comment]
Changed the name and removed usage guidelines in this PR as we are going to separate the two and add usage guidelines separately in a follow up PR.

File: docs/content/guidelines/design.mdx [hunk 8]
```diff
@@ -0,0 +1,303 @@
+---
+title: Design guidelines
+---
+
+import {Grid, Flex, Box} from '@primer/components'
+
+These are guidelines to use as a reference when designing new octicons. If you're looking for guidance on how to use octicons in your design, see our [Usage guidelines](/guidelines/usage).
```
Comment (colebemis) [inline docs/content/guidelines/design.mdx:7]:
```suggestion
These are guidelines to use as a reference when designing new octicons.
```

**Pull request merged**